### ISSUE TITLE
Fix error on attempting to claim a 19th character-specific macro

### ISSUE THREAD

**Issue by spiralofhope** [spiralofhope/SuperDuperMacro#23]
TL;DR the SDM window understand there are 18/18, so "claim" should just be greyed out..

This was discovered during exploration for #21

----

- new character
- disable addons
- `/macro`  >  "`<name>` Specific Macros"
- `/sdm`
- move the SDM window next to the macro window
- in sdm:  New
    -  Button Macro
    -  Character-specific
    -  Name:  OVERWRITE_TEST
- in the OVERWRITE_TEST macro, edit its text to be:

> TESTING

in the `/macro` list..

- switch to the "General Macros" tab, and then the "`<name>` Specific Macros" tab
- (confirm the SDM macro appears.  It cannot be clicked on)
- New > 
    -  name it "1"
    -  (don't bother choosing an icon)
    -  (leave the content empty)
- in the SDM list, for the `OVERWRITE_TEST` macro, click "Disown"
- in the `/macro` list, click the macro you called `1`
- (confirm the SDM macro disappears from the `/macro` list.
- in the SDM list, for the `OVERWRITE_TEST` macro, click "Claim"
- in the `/macro` list, click the macro you called `1`
- (confirm the SDM macro re-appears in the `/macro` list.

Continue creating macros (even with the same name) until 17 are created, the entire macros list is full.

- in the SDM list, for the `OVERWRITE_TEST` macro, click "Disown"
- in the `/macro` list, click the macro you called `1`
- (confirm the SDM macro disappears from the `/macro` list.

Create one additional macro in the `/macro` list, so that it's full.

- in the SDM list, for the `OVERWRITE_TEST` macro, click "Claim"

<details>
 <summary>Click to expand</summary>
```
1x [string "local macroIndex = sdm_GetMacroIndex(471)..."]:6: CreateMacro() failed, already have 120 macros
[C]: in function `CreateMacro'
[string "local macroIndex = sdm_GetMacroIndex(471)..."]:6: in main chunk
[C]: in function `RunScript'
...faceSuperDuperMacro\SuperDuperMacro_Core.lua:406: in function `sdm_DoOrQueue'
...faceSuperDuperMacro\SuperDuperMacro_Core.lua:167: in function `sdm_MakeBlizzardMacro'
...faceSuperDuperMacro\SuperDuperMacro_Core.lua:277: in function `sdm_SetUpMacro'
SuperDuperMacro\SuperDuperMacro_Interface.lua:93: in function <SuperDuperMacro\SuperDuperMacro_Interface.lua:91>

Locals:
(*temporary) = "OVERWRITE_TEST"
(*temporary) = "INV_MISC_QUESTIONMARK"
(*temporary) = "#sdm%&
# Enter macro text here."
(*temporary) = true
(*temporary) = 1
```
</details>
